x-pack/plugins/cases/public/common/mock/test_providers.tsx

```diff
@@ -6,50 +6,38 @@
  */
 
 import React from 'react';
-import { merge } from 'lodash';
 import { euiDarkVars } from '@kbn/ui-shared-deps-src/theme';
 import { I18nProvider } from '@kbn/i18n-react';
 import { ThemeProvider } from 'styled-components';
 import { CasesContextValue, DEFAULT_FEATURES, SECURITY_SOLUTION_OWNER } from '../../../common';
 import { CasesProvider } from '../../components/cases_context';
 import { createKibanaContextProviderMock } from '../lib/kibana/kibana_react.mock';
 import { FieldHook } from '../shared_imports';
-
 interface Props {
   children: React.ReactNode;
-  userCanCrud?: boolean;
-  features?: CasesContextValue['features'];
+  caseConfig?: Partial<CasesContextValue>;
 }
 
 window.scrollTo = jest.fn();
 const MockKibanaContextProvider = createKibanaContextProviderMock();
 
-/** A utility for wrapping children in the providers required to run most tests */
-const TestProvidersComponent: React.FC<Props> = ({
-  children,
-  userCanCrud = true,
-  features = {},
-}) => {
-  /**
-   * The empty object at the beginning avoids the mutation
-   * of the DEFAULT_FEATURES object
-   */
-  const featuresOptions = merge({}, DEFAULT_FEATURES, features);
-  return (
-    <I18nProvider>
-      <MockKibanaContextProvider>
-        <ThemeProvider theme={() => ({ eui: euiDarkVars, darkMode: true })}>
-          <CasesProvider
-            value={{ owner: [SECURITY_SOLUTION_OWNER], userCanCrud, features: featuresOptions }}
-          >
-            {children}
-          </CasesProvider>
-        </ThemeProvider>
-      </MockKibanaContextProvider>
-    </I18nProvider>
-  );
+const defaultCaseConfig = {
+  userCanCrud: true,
+  owner: [SECURITY_SOLUTION_OWNER],
+  features: DEFAULT_FEATURES,
 };
 
+/** A utility for wrapping children in the providers required to run most tests */
+const TestProvidersComponent: React.FC<Props> = ({ children, caseConfig }) => (
+  <I18nProvider>
+    <MockKibanaContextProvider>
+      <ThemeProvider theme={() => ({ eui: euiDarkVars, darkMode: true })}>
```

Comment:
I think we should merge the features with the default features like before (`merge({}, DEFAULT_FEATURES, features)`). The reason is that someone can put `caseConfig: { features: { foo: falss }}` and miss all defaults.

Reply:
sorted :D